pkg/cmd/cli/describe/chaindescriber.go

```diff
@@ -189,3 +192,46 @@ func outputHelper(info, namespace string, singleNamespace bool) string {
 	}
 	return fmt.Sprintf("<%s %s>", namespace, info)
 }
+
+// DepthFirst implements stateful depth-first graph traversal.
+// Modifies behavior of visitor.DepthFirst to allow nodes to be visited multiple
```

Comment:
I added a test for the DepthFirst traversal... can you show me a case where it would do the ininite loop?
https://github.com/csrwng/origin/commit/33be7c070309b429c24814a94ca163de1e094d46